### PR TITLE
Get Discourse Posts into Algolia search

### PR DESCRIPTION
**Pull request by JKarlavige**
## Description & motivation
Asana task: https://app.asana.com/0/0/1202844016314451/f

This creates a new Netlify function to push Discourse topics into Algolia. This function is a [Netlify Scheduled Function](https://docs.netlify.com/netlify-labs/experimental-features/scheduled-functions/).

## Preview
https://deploy-preview-2071--docs-getdbt-com.netlify.app/

## To-do before merge
- [x] Setup function on schedule
- [x] Confirm we are pushing correct Discourse data into Algolia
- [x] Switch to production Algolia env variables
- [ ] Remove `tempTopicsArr` variable and send full Discourse data to Algolia

### CODE REVIEW

File: website/functions/update-algolia.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+          tagObj.weight.position = 92
+
+          topicArr.push(tagObj)
+        })
+      } else if(group?.data?.length > 0) {
+        group.data.map(topic => {
+          if(topic?.id && topic?.slug && topic?.category_id) {
+            const topicObj = objConsts
+
+            topicObj.objectID = `discourse-${topic.category_id}-${topic.id}`
```
Comment (joellabes) [inline website/functions/update-algolia.js]: Topic IDs are unique forum-wide and a topic's category can change over time, so you may not want to include the category ID here

Comment (JKarlavige) [reply website/functions/update-algolia.js]: Ahhh good point, no need for category id if that's the case, i will adjust that back to:
```
topicObj.objectID = `discourse-${topic.id}`
```

**Comment by JKarlavige** [comment]
@john-rock & @breezyfasano I marked you for review but feel free to hold off for now. Trying to sort something out with the api keys

**Comment by JKarlavige** [comment]
Closing this out as it is not possible to implement with our current Algolia plan. We have an [Asana task](https://app.asana.com/0/1200099998847559/1203191210718277/f) to look into alternative ways we can achieve this goal.

**Pull request closed without merging**